src/libraries/pkg/test/project.csproj.template

```diff
@@ -7,13 +7,27 @@
     <TestPackageVersion>{PackageVersion}</TestPackageVersion>
     <!-- Turn off end of life target framework checks as we intentionally build older .NETCoreApp configurations. -->
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
+    <NETCoreAppMaximumVersion>99.0</NETCoreAppMaximumVersion>
 
     <SupportFilesDir Condition="'$(SupportFilesDir)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Directory.Build.props))\support</SupportFilesDir>
   </PropertyGroup>
-  
+
   <ItemGroup>
     <PackageReference Include="{PackageId}" Version="{PackageVersion}" />
   </ItemGroup>
 
+  <ItemGroup>
```

Comment:
Move this to `frameworkSettings/net6.0` as we also need to use the `RuntimeFrameworkVersion` property to test against latest `Microsoft.NETCore.App`.